src/ZPublisher/HTTPRequest.py

```diff
@@ -1441,7 +1441,7 @@ def __init__(self, fp, environ):
                 raise NotImplementedError("request parameters and body")
             if fpos is not None:
                 fp.seek(fpos)
-        elif url_qs and content_type != "application/x-www-form-urlencoded":
+        elif fp and url_qs:
```

Comment:
I suggest to replace this by `elif url_qs and fp is not None`.
`fp` is supposed to be either `None` (no body)  or a file like object (containing the body). While `bool(fp)` and `fp is not None` is equivalent for this case, the latter is more explicit.